patches/node/ci_ensure_node_tests_set_electron_run_as_node.patch

```diff
@@ -0,0 +1,72 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Charles Kerr <charles@charleskerr.com>
+Date: Mon, 7 Aug 2023 13:12:29 -0500
+Subject: ci: ensure node tests set ELECTRON_RUN_AS_NODE=1
+
+Some node tests / test fixtures spawn other tests that clobber env,
```

Comment:
This isn't upstreamable in its current state, but i think it could be! I wouldn't necessarily block on it for this PR, but we can probably address this with something more like this:

<details><summary>Diff Suggestion</summary>
<p>

```diff
diff --git a/test/common/assertSnapshot.js b/test/common/assertSnapshot.js
index 83ee45f5f9..2bc7f0092c 100644
--- a/test/common/assertSnapshot.js
+++ b/test/common/assertSnapshot.js
@@ -62,7 +62,7 @@ async function spawnAndAssert(filename, transform = (x) => x, { tty = false, ...
   const flags = common.parseTestFlags(filename);
   const executable = tty ? 'tools/pseudo-tty.py' : process.execPath;
   const args = tty ? [process.execPath, ...flags, filename] : [...flags, filename];
-  if (options && options.env) options.env.ELECTRON_RUN_AS_NODE = 1;
+  if (options?.env) options.env = { ...process.env, ...options.env };
   const { stdout, stderr } = await common.spawnPromisified(executable, args, options);
   await assertSnapshot(transform(`${stdout}${stderr}`), filename);
 }
diff --git a/test/fixtures/test-runner/output/arbitrary-output-colored.js b/test/fixtures/test-runner/output/arbitrary-output-colored.js
index dd8296e631..0dbaff42d0 100644
--- a/test/fixtures/test-runner/output/arbitrary-output-colored.js
+++ b/test/fixtures/test-runner/output/arbitrary-output-colored.js
@@ -6,6 +6,6 @@ const fixtures = require('../../../common/fixtures');
 
 (async function run() {
   const test = fixtures.path('test-runner/output/arbitrary-output-colored-1.js');
-  await once(spawn(process.execPath, ['--test', test], { stdio: 'inherit', env: { ELECTRON_RUN_AS_NODE: 1, FORCE_COLOR: 1 } }), 'exit');
-  await once(spawn(process.execPath, ['--test', '--test-reporter', 'tap', test], { stdio: 'inherit', env: { ELECTRON_RUN_AS_NODE: 1, FORCE_COLOR: 1 }  }), 'exit');
+  await once(spawn(process.execPath, ['--test', test], { stdio: 'inherit', env: { ...process.env, FORCE_COLOR: 1 } }), 'exit');
+  await once(spawn(process.execPath, ['--test', '--test-reporter', 'tap', test], { stdio: 'inherit', env: { ...process.env, FORCE_COLOR: 1 }  }), 'exit');
 })().then(common.mustCall());
\ No newline at end of file
diff --git a/test/parallel/test-node-output-console.mjs b/test/parallel/test-node-output-console.mjs
index efca7811dc..5a1b9feb6c 100644
--- a/test/parallel/test-node-output-console.mjs
+++ b/test/parallel/test-node-output-console.mjs
@@ -31,7 +31,6 @@ describe('console output', { concurrency: true }, () => {
     .transform(snapshot.replaceWindowsLineEndings, snapshot.replaceWindowsPaths, replaceStackTrace);
   for (const { name, transform, env } of tests) {
     it(name, async () => {
-      if (env) env.ELECTRON_RUN_AS_NODE = 1;
       await snapshot.spawnAndAssert(fixtures.path(name), transform ?? defaultTransform, { env });
     });
   }
diff --git a/test/parallel/test-node-output-errors.mjs b/test/parallel/test-node-output-errors.mjs
index b9a55fb7ea..fca2149fea 100644
--- a/test/parallel/test-node-output-errors.mjs
+++ b/test/parallel/test-node-output-errors.mjs
@@ -53,7 +53,6 @@ describe('errors output', { concurrency: true }, () => {
     !skipForceColors ? { name: 'errors/force_colors.js', env: { FORCE_COLOR: 1 } } : null,
   ].filter(Boolean);
   for (const { name, transform, env } of tests) {
-    if (env) env.ELECTRON_RUN_AS_NODE = 1;
     it(name, async () => {
       await snapshot.spawnAndAssert(fixtures.path(name), transform ?? defaultTransform, { env });
     });
```

</p>
</details>

Reply:
I like the idea of changing this into some upstreamable form, but unconditionally inheriting all of process.env seems problematic -- I can think of plenty of use cases for a test to intentionally clobber env.

Maybe there's some other variation that doesn't involve hardcoding `ELECTRON_RUN_AS_NODE`, e.g. passing `tools/test.py` a list of env vars to enherit?